Add methods to `Descriptor` + `DescriptorPublicKey`

reez opened the pull request:
<!-- Erase any parts of this template not applicable to your Pull Request. -->

### Description

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.DescriptorPublicKey.html#method.master_fingerprint

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.DescriptorPublicKey.html#method.is_multipath

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.Descriptor.html#method.is_multipath

https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.Descriptor.html#method.into_single_descriptors

**_Would love for this to get into our beta6_**

### Notes to the reviewers

Use Cases:
- creating a wallet from a descriptor having the ability to check if it's multipath or not
- creating a wallet from an xpub having the ability to construct a DescriptorPublicKey fromString then get the fingerprint

### Changelog notice

<!-- Notice the release manager should include in the release tag message changelog -->
<!-- See https://keepachangelog.com/en/1.0.0/ for examples -->

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [x] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

rustaceanrob commented:
Concept ACK

Cool additions

rustaceanrob commented on bdk-ffi/src/descriptor.rs:
```diff
@@ -266,6 +266,10 @@ impl Descriptor {
         descriptor.to_string_with_secret(key_map)
     }
+
+    pub(crate) fn is_multipath(&self) -> bool {
```
Just out of curiosity, why are all of these access modifiers `crate` level? Clearly has no effect on the bindings but seems like a scope definition that doesn't actually apply

reez replied:
good question/point-

reez commented:
> Concept ACK
> 
> Cool additions

Thanks!

reez commented:
I get a failure because `cargo clippy` doesn't like the naming convention, I think the best option to get this to pass are renaming:

```
    pub(crate) fn to_single_descriptors(&self) -> Result<Vec<Arc<Descriptor>>, MiniscriptError> {
    }
```

will do that right now just cuz I would like this to pass even though that naming isn't exactly equivalent to the function we are wrapping.

reez commented on bdk-ffi/src/bdk.udl:
```diff
@@ -549,6 +590,12 @@ interface DescriptorPublicKey {
 
   string as_string();
+
+/// Whether or not this key has multiple derivation paths.
```
yep sorry didn't update after comment on https://github.com/bitcoindevkit/bdk-ffi/pull/615#pullrequestreview-2437116595 but will update now-

reez commented:
> Just a tiny comment on the identation of the docstrings, but otherwise this is good to go IMO.

Updated docstrings indentation, rebased, ready to merge 👍

thunderbiscuit commented:
Sorry man I merged #629 before this one, so you'll have to rebase, but feel free to merge afterwards!
